app/bundles/CoreBundle/DependencyInjection/Compiler/ShortenerServicePass.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Mautic\CoreBundle\DependencyInjection\Compiler;
+
+use Mautic\CoreBundle\Shortener\Shortener;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class ShortenerServicePass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition(Shortener::class)) {
+            return;
+        }
+
+        $storageDef = $container->findDefinition(Shortener::class);
+        foreach ($container->findTaggedServiceIds('mautic.shortener.service') as $id => $tags) {
+            $storageDef->addMethodCall('addService', [$id, new Reference($id)]);
```

Comment:
Then this could be simplified to 
```suggestion
            $storageDef->addMethodCall('addService', [new Reference($id)]);
```

Reply:
These changes make choices in configuration weird.  Should the service name for the translation part of ShortenerServiceInterface?

![image](https://github.com/mautic/mautic/assets/462477/0b4e6c03-c5d1-4f78-b037-024ffd423c8c)

Reply:
I wouldn't mind if the value would be the FQCN. It's not optimal, but it would work fine. But you are right. The option name should be more user friendly. I think that adding a new method `getName()` to the `ShortenerServiceInterface` would be probably the best option.

Reply:
or `getTranslatableKey()`? So devs would know that it will be the translation key that will be pushed through the translator.

Reply:
getTranslatableNameKey seems weird to me
probably getPublicName we use the same in integration. And we can let devs choose If they use the global name or translatable string.